pkg/registry/apis/provisioning/resources/replicator.go

```diff
@@ -99,6 +104,39 @@ func (r *Replicator) Sync(ctx context.Context) (string, error) {
 	return latest, nil
 }
 
+func (r *Replicator) Unsync(ctx context.Context, opts UnsyncOptions) error {
```

Comment:
Is  Unsync == Delete everything we added from this repo?

Reply:
I was thinking that this method will do everything needed to "detach" a repository from Grafana after is deleted. For now, I plan to: 
- Clear repo info from folder.
- Remove folder itself **(optional)**.
- Remove folder contents **(optional)**.

Maybe `Shutdown`? `TearDown`?